### PR TITLE
WOOC-714 Block the BO API from public access

### PR DESCRIPTION
**Pull request by issam-payplug**

### CODE REVIEW

File: src/Admin/Vue.php [hunk 1]
```diff
@@ -27,6 +27,9 @@ public function init() {
 			$logged = $this->payplug_section_logged();
 			$payplug_wooc_settings = get_option( 'woocommerce_payplug_settings', [] );
 			unset($payplug_wooc_settings["payplug_live_key"]);
+			unset($payplug_wooc_settings["payplug_test_key"]);
+			unset($payplug_wooc_settings["payplug_password"]);
+			unset($payplug_wooc_settings["payplug_merchant_id"]);
```
Comment (GhDj) [inline src/Admin/Vue.php:32]: we need the merchant id for the one click and save card

File: src/Admin/Vue.php [hunk 2]
```diff
@@ -27,6 +27,9 @@ public function init() {
 			$logged = $this->payplug_section_logged();
 			$payplug_wooc_settings = get_option( 'woocommerce_payplug_settings', [] );
 			unset($payplug_wooc_settings["payplug_live_key"]);
+			unset($payplug_wooc_settings["payplug_test_key"]);
```
Comment (GhDj) [inline src/Admin/Vue.php:30]: we need the test key for inactive merchants

File: src/Admin/Ajax.php [hunk 3]
```diff
@@ -35,66 +35,71 @@ class Ajax {
...
 				'methods' => 'POST',
 				'callback' => [ $this, 'api_check_oney_permissions' ],
-				'permission_callback' => '__return_true'
+				'permission_callback' => function () use ($permission)  {return $permission ;}
 			) );
 		});
 
 	}
 
+	public function admin_loggedin() {
```
Comment (GhDj) [inline src/Admin/Ajax.php]: unused function

**Review by GhDj — CHANGES_REQUESTED** [review]

File: src/Admin/Vue.php (hunk 1 above)
Comment (issam-payplug) [reply src/Admin/Vue.php:32]: This will prevent sending the mercahnt _id to vue, and it is not used there.

File: src/Admin/Vue.php (hunk 2 above)
Comment (issam-payplug) [reply src/Admin/Vue.php:30]: Same thing as merchant_id, payplug_test_key is not used in vue side, only used in server side

File: src/Admin/Ajax.php (hunk 3 above)
Comment (issam-payplug) [reply src/Admin/Ajax.php]: Removed

Comment (issam-payplug) [reply src/Admin/Ajax.php]: Removed

**Comment by GhDj** [comment]
we also should add 'show_in_index' => false to every register_rest_route so our api can't be listed

**Review by GhDj — APPROVED** [review]

**Pull request merged**